test/util/all_close_f.cpp

```diff
@@ -203,6 +311,82 @@ bool test::all_close_f(const vector<float>& a,
     return rc;
 }
 
+bool test::all_close_f(const vector<double>& a, const vector<double>& b, int tolerance_bits)
+{
+    constexpr int mantissa_bits = 53;
+
+    bool rc = true;
+    if (a.size() != b.size())
+    {
+        throw ngraph_error("a.size() != b.size() for all_close_f comparison.");
+    }
+    vector<uint64_t> distances = float_distances(a, b);
+
+    // e.g. for double with 52 bit mantissa, 2 bit accuracy, and hard-coded 11 bit exponent_bits
+    // tolerance_bit_shift = 64 -           (1 +  11 + (53 -     1         ) - 2             )
+    //                       double_length   sign exp   mantissa implicit 1    tolerance_bits
+    uint64_t tolerance_bit_shift = 64 - (1 + 11 + (mantissa_bits - 1) - tolerance_bits);
+    uint64_t tolerance = static_cast<uint64_t>(1U) << tolerance_bit_shift;
+    uint64_t max_distance = 0;
+    uint64_t min_distance = ULLONG_MAX;
+    size_t max_distance_index = 0;
+    size_t min_distance_index = 0;
+    size_t diff_count = 0;
+    for (size_t i = 0; i < a.size(); ++i)
+    {
+        if (distances[i] > max_distance)
+        {
+            max_distance = distances[i];
+            max_distance_index = i;
+        }
+        if (distances[i] < min_distance)
+        {
+            min_distance = distances[i];
+            min_distance_index = i;
+        }
+        bool is_close_f = distances[i] <= tolerance;
```

Comment:
note here: Discussed with @gcwenger, is it possible to use ` <` instead of `<=`, currently when set tolerance bit to 0, it will still pass if there is 1 ulp diff.